release_notes/models.py

```diff
@@ -0,0 +1,210 @@
+import dataclasses
+import re
+import typing
+
+import gci.componentmodel
+import git
+import github3.pulls
+from git import Commit
+
+import cnudie.util
+
+
+@dataclasses.dataclass
+class Author:
+    # for pull requests
+    username: str  # the GitHub username
+
+    # for commits
+    display_name: str  # the name which authored the commit
+    email: str  # the email the commit was authored
+
+    def __str__(self) -> str:
+        if self.username and self.username.strip():
+            return f"@{self.username}"
+        if self.display_name and self.display_name.strip():
+            return f"`{self.display_name} <{self.email}>`"
+        return ""
+
+
+def author_from_commit(commit: git.Commit) -> Author:
+    return Author("", commit.author.name, commit.author.email)
+
+
+def author_from_pull_request(pull_request: github3.pulls.ShortPullRequest) -> Author:
+    return Author(pull_request.user.login, "", "")
+
+
+#
+
+@dataclasses.dataclass
+class ReferenceType:
+    identifier: str  # identifier for release note block
+    prefix: str  # prefix for generated release notes
+
+
+REF_TYPE_PULL = ReferenceType(identifier="#", prefix="#")
+REF_TYPE_COMMIT = ReferenceType(identifier="$", prefix="@")
+
+REF_TYPES = [REF_TYPE_PULL, REF_TYPE_COMMIT]
```

Comment:
nit: rather use tuples

I did not check: if not used outside this module, please use lowercase + prefix w/ underscore